client/src/modules/patcher.js

```
@@ -0,0 +1,93 @@
+/**
+ * BetterDiscord Component Patcher
+ * Copyright (c) 2015-present JsSucks - https://github.com/JsSucks
+ * All rights reserved.
+ * https://github.com/JsSucks - https://betterdiscord.net
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree. 
+*/
+
+import WebpackModules from './webpackmodules';
+
+export default class Patcher {
+    static get patches() { return this._patches || (this._patches = {}) }
+    static resolveModule(mn) {
+        if (mn instanceof Function || (mn instanceof Object && !(mn instanceof Array))) return mn;
+        if ('string' === typeof mn) return WebpackModules.getModuleByName(mn);
+        if (mn instanceof Array) return WebpackModules.getModuleByProps(mn);
+        return null;
+    }
+    static overrideFn(patch) {
+        return function () {
+            for (const s of patch.supers) {
+                try {
+                    s.fn.apply(this, arguments);
+                } catch (err) { }
```

silently suppressing is bad! at least add `console.warn` otherwise it would be very hard to find errors

---

Temporary